### ISSUE TITLE
Comments on JOSS paper

### ISSUE THREAD

**Issue by matt-graham** [UncertaintyInComplexSystems/bamojax#1]
Raising as part of openjournals/joss-reviews#8642

The paper reads very well in general. Some comments and suggestions

- [ ] While you spell out _Markov chain Monte Carlo_ in full on the first usage in line 10 you don't currently define the shortened acronym form MCMC alongside it. 
- [ ] Similarly _Gaussian process_ needs GP acronym define alongside it in line 27. 
- [ ] NUTS acronym should be expanded to full _No U-Turn Sampler_ and a reference to [Hoffman and Gelman (2014)](https://jmlr.org/papers/v15/hoffman14a.html) added as its only used once and readers will not necessarily be familiar with acronym.
- [ ] _logdensity function_ in line 24 needs a space between _log_ and _density_), and while the shortened usage is common I would say _logarithm of the probability density function_ might be more explicit for readers less familiar with the area.
- [ ] Nitpick, but it would be good to change the instances of _Blackjax_ to _BlackJAX_ to be consistent with the capitalisation most commonly used by the authors of the package (specifically in the cited paper) and change the instance in fixed-width font in line 51 to be in plain style to be consistent with how package names are rendered elsewhere.
- [ ] Another nitpick - in line 31 in _prior and likelihood densities_ - technically the likelihood is not a probability density function as a function of the parameters (as it the conditional density of observed data given parameters under model). _prior and likelihood terms_ might be better.
- [ ] The claim that _elliptical slice sampling for the GP is often more efficient than applying NUTS to all variables jointly_,  I think either needs a reference supporting it added or some evidence given. It would also be good to reference [Murray, Adams and MacKay (2010)](https://proceedings.mlr.press/v9/murray10a) when mentioning elliptical slice sampling. While I believe elliptical slice sampling can be more efficient (in terms of number of effective samples per log density / gradient evaluation) than Hamiltonian Monte Carlo methods like NUTS in Gaussian process models, it is very dependent on the specific problem and parameterization used.
- [ ] In the comparison to existing tools - I think it would be good to mention a few other related tools here. Specifically [NumPyro](https://num.pyro.ai/en/stable/index.html) and [Oryx](https://github.com/jax-ml/oryx) are both alternative JAX based probabilistic programming languages. NumPyro specifically supports [combining Gibbs sampling for some variables with Hamiltonian Monte Carlo updates for other variables](https://num.pyro.ai/en/stable/mcmc.html). Oryx as far as I can tell doesn't provide out-of-the-box support for Gibbs sampling, though its ability to [construct functions for conditionals of joint distribution defined in its PPL](https://www.tensorflow.org/probability/oryx/api_docs/python/oryx/core/ppl/conditional) means it's probably possible with a bit of work.
- [ ] The statement that _PyMC ... can also interface with Blackjax for inference, this only supports using a single log-density function that describes the entire probabilistic model. This precludes Gibbs sampling, where individual model parameters are updated in turn, which in practice can greatly increase the efficiency of approximate inference_ is I think a bit misleading. While the example in the BlackJAX documentation of [how to interface with a PyMC model](https://blackjax-devs.github.io/blackjax/examples/howto_use_pymc.html) does pass a log density function defined on all random variables in the model, this could still easily be used to sample just a subset of the parameters by creating a partial function that fixes some values of the parameters. Further, the PyMC BlackJAX integration in the [`pymc.sampling.jax.sample_blackjax_nuts`](https://www.pymc.io/projects/docs/en/stable/api/generated/pymc.sampling.jax.sample_blackjax_nuts.html#) function specifically allows specifying a `var_names` argument to apply it to a subset of variables in the model, and could be combined with other PyMC transition kernels for different variables, including Gibbs like steps. It looks like this wouldn't be particularly easy to do and might require a certain amount of boilerplate but I would say it's not _precluded_.

**Comment by mhinne** [comment]
Thank you for the helpful comments! The textual mistakes I have updated right away.

I agree that the statements on comparing bamojax to the mentioned tools may have been a bit overzealous. I have rewritten the comparison section somewhat to better reflect the aim of bamojax (allowing for all the fast Blackjax algorithms but also stay close to JAX Numpy for method development).

Please let me know if you agree with these changes.
